lib/posix/getopt/Kconfig

```diff
@@ -4,6 +4,7 @@
 
 menuconfig GETOPT
 	bool "Getopt library support"
+	default y if POSIX_API
```

Comment:
Any reason this should not `depends on POSIX_API`?

Reply:
`CONFIG_GETOPT` is mainly there just to say to compile / link with getopt.c / .o .

In that sense, `CONFIG_POSIX_API=y` is a shortcut to enable all POSIX features, but people can choose to explicitly build-in only specific POSIX features. Of course, in that case `CONFIG_POSIX_API` needs to be `n`, and each specific feature must be manually selected.